app/restore_to.go

```diff
@@ -0,0 +1,145 @@
+package app
+
+import (
+	"fmt"
+	"os/exec"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/rancher/longhorn-engine/replica"
+	"github.com/rancher/longhorn-engine/util"
+	"github.com/urfave/cli"
+)
+
+const DefaultImageFormat = "qcow2"
+
+var SupportedImageFormats = []string{
+	"rbd",
+	"host_cdrom",
+	"blkdebug",
+	"qcow",
+	"host_device",
+	"vpc",
+	"qcow2",
+	"cloop",
+	"vdi",
+	"sheepdog",
+	"qed",
+	"nbd",
+	"tftp",
+	"vvfat",
+	"ftp",
+	"ftps",
+	"https",
+	"dmg",
+	"http",
+	"vmdk",
+	"iscsi",
+	"parallels",
+	"raw",
+	"bochs",
+	"quorum",
+	"null-aio",
+	"null-co",
+	"vhdx",
+	"blkverify",
+	"file",
+}
+
+func RestoreToCmd() cli.Command {
+	return cli.Command{
+		Name: "restore-to",
+		Flags: []cli.Flag{
+			cli.StringFlag{
+				Name:  "backing-file",
+				Usage: "filepath or dirpath containing exactly one qcow2 backing file",
+			},
+			cli.StringFlag{
+				Name:  "backup-url",
+				Usage: "backup URL to be published",
+			},
+			cli.StringFlag{
+				Name:  "image-format",
+				Usage: "format of image to produce",
+				Value: DefaultImageFormat,
+			},
+			cli.StringFlag{
+				Name:  "output-file",
+				Usage: "filepath to write the resulting image to",
+				Value: "/image/base.qcow2",
+			},
+		},
+		Action: func(c *cli.Context) {
+			if err := restoreTo(c); err != nil {
+				logrus.Fatalf("Error running restore to file command: %v", err)
+			}
+		},
+	}
+}
+
+func imageFormatSupported(desiredFormat string) bool {
+	for _, supportedFormat := range SupportedImageFormats {
+		if desiredFormat == supportedFormat {
+			return true
+		}
+	}
+	return false
+}
+
+func restoreTo(c *cli.Context) error {
+	if c.NArg() != 1 {
+		return fmt.Errorf("directory name is required")
+	}
+	dir := c.Args()[0]
+
+	backingFile, err := openBackingFile(c.String("backing-file"))
+	if err != nil {
+		return err
+	}
+	s := replica.NewServer(dir, backingFile, 512)
```

Comment:
You should only need [this](https://github.com/rancher/longhorn-engine/pull/325/commits/df56c7e4735890aceb5351da93fef08569b3bf65#diff-09f0fd9605d3f3cf6757ff3d4d30fc9aR358)(along with some precheck and unescape, see the logic) directly from backupstore API. You don't need to use replica server.